### PR TITLE
[various] Allow resuming pending authorization after activity recreation on Android

### PR DESCRIPTION
**Pull request by moritz-j**
We have encountered an issue where the auth response is not handled after returning from the browser back to the app. This happens when the app's Activity is killed by Android (e.g. for clearing up memory). This PR adds a resumePendingAuthorization() method that allows the app to get the "missed" auth response after reinitialising itself.

### Steps to reproduce:
1. Enable "Don't keep activities" in Android developer settings.
2. Start auth flow.
3. Complete authentication.

=>  Browser closes, but app doesn't receive the auth response.

### Behaviour with fix:
1. Browser auth flow started
2. Activity killed in background
3. Browser flow finishes
4. Activity is recreated and flutter reinitialized
5. App calls resumePendingAuthorization() and receives response
6. User logged in

### CODE REVIEW

**Review by MaikuB — CHANGES_REQUESTED** [review]
Thanks for the PR. Can you update the example app so it shows how this could be used? This includes adding code comments to the example app where appropriate

File: flutter_appauth/android/src/main/java/io/crossingthestreams/flutterappauth/FlutterAppauthPlugin.java [hunk 1]
```diff
@@ -88,6 +89,7 @@ public class FlutterAppauthPlugin
   private Context applicationContext;
   private Activity mainActivity;
   private PendingOperation pendingOperation;
+  private PendingAuthorization pendingAuthorization;
```
[the remark this answers is not in the document]
Comment (moritz-j) [reply flutter_appauth/android/src/main/java/io/crossingthestreams/flutterappauth/FlutterAppauthPlugin.java:92]: The value is not lost. Only the flutter engine is recreated, not the native host activity. Manual testing also confirmed this. Trying to have the auth result survive the whole process being killed would require more effort while I don't know of any real use case that would benefit from this. I would file this under accepted behaviour.

File: flutter_appauth/android/src/main/java/io/crossingthestreams/flutterappauth/FlutterAppauthPlugin.java [hunk 2]
```diff
@@ -198,6 +200,17 @@ public void onMethodCall(MethodCall call, @NonNull Result result) {
...
         }
         break;
+      case RESUME_PENDING_AUTHORIZATION_METHOD:
+        try {
+          handleResumePendingAuthorizationMethodCall(result);
+        } catch (Exception ex) {
+          String errorCode = AUTHORIZE_ERROR_CODE;
+          if (pendingAuthorization != null && pendingAuthorization.exchangeCode ) {
+            errorCode = AUTHORIZE_AND_EXCHANGE_CODE_ERROR_CODE;
+          }
```
[the remark this answers is not in the document]
Comment (moritz-j) [reply flutter_appauth/android/src/main/java/io/crossingthestreams/flutterappauth/FlutterAppauthPlugin.java]: This seems like a valid finding of an edge case bug. I have moved the error handling to the method that consumes the pendingAuthorization so the correct error code can always be used.

File: flutter_appauth_platform_interface/test/method_channel_flutter_appauth_test.dart [hunk 3]
```diff
@@ -190,4 +190,92 @@ void main() {
...
+    test(
+        'returns an AuthorizationTokenResponse for a resumed '
+        'authorizeAndExchangeCode() flow', () async {
+      TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger
+          .setMockMethodCallHandler(channel, (MethodCall methodCall) async {
+        log.add(methodCall);
+        return <Object, Object?>{
+          'accessToken': 'someAccessToken',
+          'refreshToken': 'someRefreshToken',
+          'accessTokenExpirationTime': 1784239200000,
```
[the remark this answers is not in the document]
Comment (moritz-j) [reply flutter_appauth_platform_interface/test/method_channel_flutter_appauth_test.dart:255]: Days since time zone issue: -1

**Comment by moritz-j** [comment]
Hey @MaikuB, thanks for the review. I have added this feature to the example app and fixed 2/3 Copilot findings (the 3rd being not really useful).
